File: g-b.cpp
// using opencv

#include <opencv2/opencv.hpp>
#include <iostream>

int main(){
    cv::Mat image = cv::imread( "input.jpg" );
    if( image.empty() ){
        std::cout << "Error: Could not load image!" << std::endl;
        return -1;
    }
    
    cv::Mat blurred;
    cv::GaussianBlur( image, blurred, cv::Size( 3, 3 ), 0 );
    cv::imwrite( "output.jpg", blurred );
    
    std::cout << "Gaussian blur applied!" << std::endl;
    return 0;
}

File: g-b.py
# using opencv

import cv2

def gaussian_blur( image_path, output_path, kernel_size=( 3,3 ) ):
    image = cv2.imread( image_path )
    blurred = cv2.GaussianBlur( image, kernel_size, 0 )
    cv2.imwrite( output_path, blurred )

gaussian_blur( "input.jpg", "output.jpg" )
print( "Gaussian blur applied!" )
